mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala

```diff
@@ -179,7 +179,7 @@ abstract class GeneralizedLinearAlgorithm[M <: GeneralizedLinearModel]
    */
   def run(input: RDD[LabeledPoint]): M = {
     if (numFeatures < 0) {
-      numFeatures = input.map(_.features.size).first()
+      numFeatures = input.first().features.size
```

Comment:
There is also a good possibility that input is very large. Is accessing the first element of a RDD slower than doing a distributed operation across the entire RDD?